pilot/cmd/pilot-agent/options/agent.go

```diff
@@ -23,22 +23,30 @@ import (
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pkg/bootstrap/platform"
 	istioagent "istio.io/istio/pkg/istio-agent"
+	"istio.io/istio/pkg/util/sets"
+	"istio.io/istio/pkg/wasm"
 )
 
 // Similar with ISTIO_META_, which is used to customize the node metadata - this customizes extra header.
 const xdsHeaderPrefix = "XDS_HEADER_"
 
 func NewAgentOptions(proxy *model.Proxy, cfg *meshconfig.ProxyConfig) *istioagent.AgentOptions {
 	o := &istioagent.AgentOptions{
-		XDSRootCerts:                xdsRootCA,
-		CARootCerts:                 caRootCA,
-		XDSHeaders:                  map[string]string{},
-		XdsUdsPath:                  filepath.Join(cfg.ConfigPath, "XDS"),
-		IsIPv6:                      proxy.IsIPv6(),
-		ProxyType:                   proxy.Type,
-		EnableDynamicProxyConfig:    enableProxyConfigXdsEnv,
-		EnableDynamicBootstrap:      enableBootstrapXdsEnv,
-		WASMInsecureRegistries:      strings.Split(wasmInsecureRegistries, ","),
+		XDSRootCerts:             xdsRootCA,
+		CARootCerts:              caRootCA,
+		XDSHeaders:               map[string]string{},
+		XdsUdsPath:               filepath.Join(cfg.ConfigPath, "XDS"),
+		IsIPv6:                   proxy.IsIPv6(),
+		ProxyType:                proxy.Type,
+		EnableDynamicProxyConfig: enableProxyConfigXdsEnv,
+		EnableDynamicBootstrap:   enableBootstrapXdsEnv,
+		WASMOptions: wasm.Options{
```

Comment:
You can add a construction function for Options, like NewOption() to replace `defaultOptions`

And then you can not hardcoding the default value in https://github.com/istio/istio/pull/39050/files#diff-be623e52351ff6a4def385a3dce45551df39e74750e7625dc937ae7a1aa1c8bcR107-R117

Reply:
Just move the values into the constant to avoid hard coding in the two places.